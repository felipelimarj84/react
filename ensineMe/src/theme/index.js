const theme = {
  palette: {
    primary: {
      main: "#7f3f98",
    },
    secondary: {
      main: "#17243c",
      light: "#bbcce9",
    },
    neutral: {
      darkest: "#000000",
      darker: "#333333",
      dark: "#4d4d4d",
      medium: "#808080",
      light: "#bfbfbf",
      lighter: "#f2f2f2",
      lightest: "#ffffff",
    },
    success: {
      dark: "#147c38",
      light: "#e7f2eb",
    },
    error: {
      dark: "#df402b",
      light: "#fcecea",
    },
    alert: {
      dark: "#ddac00",
      light: "#fcf7e5",
    },
    information: {
      dark: "#2f80ed",
      light: "#eaf2fd",
    },
  },
};

export default theme;
