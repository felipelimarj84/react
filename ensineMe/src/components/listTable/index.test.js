import React from "react";
import ReactDOM from "react-dom";
import ListTable from "./index";
import { ThemeProvider } from "styled-components";
import theme from "../../theme";
import "jest-styled-components";

// import Enzyme, { shallow, render, mount } from 'enzyme';
// import toJson from 'enzyme-to-json';
// import Adapter from 'enzyme-adapter-react-16';
import { cleanup, render, fireEvent } from "@testing-library/react";

// Enzyme.configure({ adapter: new Adapter() })

afterEach(cleanup);

it("<ListTable/> matches snapshot basic", () => {
  const component = render(
    <ThemeProvider theme={theme}>
      <ListTable
        info={[{ teste: "teste1", teste2: "teste2", teste3: "teste3" }]}
        header={[
          { nome: "bonito", ref: "teste" },
          { nome: "bonitos", ref: "teste2" },
          { nome: "bonitos", ref: "teste3", buttons: [{ title: "teste" }] },
        ]}
      />
    </ThemeProvider>
  );

  expect(component.container).toMatchSnapshot();
});
