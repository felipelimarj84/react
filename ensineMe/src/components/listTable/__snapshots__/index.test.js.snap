// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListTable/> matches snapshot basic 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 20px;
  width: 100%;
  font-size: 2rem;
  padding: 10px;
}

.c0 .info-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  gap: 15px;
}

.c0 .info-container .grid-container {
  display: grid;
  width: 100%;
  grid-template-columns: repeat(3,1fr);
}

.c0 .info-container .title-item {
  border-bottom: 2px solid #ddd;
  padding: 5px;
}

.c0 .info-container .item-list {
  border-bottom: 1px solid #ddd;
  padding: 5px;
  font-size: 1.8rem;
}

.c0 .info-container .item-list .item {
  font-size: 1.8rem;
}

.c0 .info-container .item-list .item .button-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 15px;
}

.c1 {
  padding: 10px;
  border-radius: 5px;
  font-size: 2.5rem;
  background-color: #7f3f98;
  color: #bfbfbf;
}

<div>
  <div
    class="c0"
  >
    <div
      class="info-container"
    >
      <div
        class="grid-container title-item"
      >
        <p>
          bonito
        </p>
        <p>
          bonitos
        </p>
        <p>
          bonitos
        </p>
      </div>
      <div
        class="grid-container item-list"
      >
        <div
          class="item"
        >
          <p>
            teste1
          </p>
        </div>
        <div
          class="item"
        >
          <p>
            teste2
          </p>
        </div>
        <div
          class="item"
        >
          <p>
            teste3
          </p>
          <div
            class="button-container"
          >
            <button
              class="c1"
            >
              teste
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
